spec.bs

```diff
@@ -1738,6 +1754,52 @@ Several APIs specific to fenced frames are defined on the {{Fence}} interface.
   </wpt>
 </div>
 
+<div algorithm>
+  The <dfn method for=Fence>disableUntrustedNetwork()</dfn> method steps are:
+
+  1. Let |p| be [=a new promise=].
+
+  1. Let |instance| be [=this=]'s [=relevant global object=]'s [=Window/browsing context=]'s
+     [=browsing context/fenced frame config instance=].
+
+  1. If |instance| is null, then [=resolve=] |p| with {{undefined}} and return |p|.
+
+  1. If |instance|'s [=fenced frame config instance/can disable untrusted network=] is false, then
+     [=resolve=] |p| with {{undefined}} and return |p|.
+
+  1. If the [=relevant settings object=]'s [=environment settings object/origin=] and |instance|'s
+     [=fenced frame config instance/mapped url=]'s [=url/origin=] are not [=same origin=], then
+     [=resolve=] |p| with {{undefined}} and return |p|.
+
+  1. Let |global| be [=this=]'s [=relevant global object=].
+
+  1. Run the following steps [=in parallel=]:
+
+     1. Let |fencedFrameNonce| be |instance|'s [=fenced frame config instance/partition nonce=].
+
+     1. Let |credentiallessNonce| be
+
+     Issue: the page credentialless nonce
+     (<a href="https://github.com/WICG/fenced-frame/issues/151">WICG/fenced-frame#151</a>)
+
+     1. Revoke the partition nonces |fencedFrameNonce| and |credentiallessNonce|.
+
+     Issue: Spec the revocation process.
+     (<a href="https://github.com/WICG/fenced-frame/issues/151">WICG/fenced-frame#151</a>)
+
+     1. Set |instance|'s [=fenced frame config instance/has disabled untrusted network=] to true.
+
+     1. Wait on all nested fenced frames to disable network too.
+
+     Issue: Spec this waiting more formally.
+     (<a href="https://github.com/WICG/fenced-frame/issues/151">WICG/fenced-frame#151</a>)
```

Comment:
This is a HUGE TODO, so I'm not really comfortable landing this PR without those fundamental mechanisms spec'ed out a bit more rigorously, just FYI. "Wait on all nested fenced frames to disable network too" is pretty rough, so I'd like to have a good draft of how this is done in hand before landing this PR.

Reply:
I have created a draft PR https://github.com/WICG/fenced-frame/pull/176 to spec this more formally. That should hopefully unblock this PR.

Reply:
Hey y'all, I'm likely going to pick up the network revocation spec work from gtanzer, since he's transitioned to other work. This seems like the first major PR to have in place, and this is the last outstanding comment. Is the draft PR #176 sufficient scaffolding for disabling network in nested frames? If so, I'll do whatever else is needed to get this PR over the finish line.

Reply:
I'd like to get https://github.com/WICG/fenced-frame/pull/176 just a tiny bit further along before merging this, if that's alright. It's in decent shape, I just want to be more confident in its direction before proceeding.